### PR TITLE
search contexts: community pages

### LINKED ISSUE #24273: Convert Cloud repogroups to search contexts
Create instance-level search contexts from existing Cloud repogroups (found in global settings).

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #24273

Before this PR is merged we will have to:
✅  Create the corresponding search contexts (cncf, stanford, etc.) on Cloud.
✅  Update handbook to describe how to add new community pages (https://about.sourcegraph.com/handbook/marketing/oss_community_pages): https://github.com/sourcegraph/about/pull/4808

### CODE REVIEW

File: client/web/src/search/home/SearchPage.tsx [hunk 1]
```diff
@@ -98,9 +96,9 @@ export const SearchPage: React.FunctionComponent<SearchPageProps> = props => {
                 />
             </div>
             <div className="flex-grow-1">
-                {props.isSourcegraphDotCom &&
-                    props.showRepogroupHomepage &&
-                    (!props.authenticatedUser || !props.showEnterpriseHomePanels) && <LoggedOutHomepage {...props} />}
+                {props.isSourcegraphDotCom && (!props.authenticatedUser || !props.showEnterpriseHomePanels) && (
```
Comment (limitedmage) [inline client/web/src/search/home/SearchPage.tsx:99]: I don't think we should show the logged out homepage if the user is logged in but home panels are disabled. We should show an empty page instead.

```suggestion
                {props.isSourcegraphDotCom && (!props.authenticatedUser) && (
```

File: client/web/src/routes.tsx [hunk 2]
```diff
@@ -252,41 +245,7 @@ export const routes: readonly LayoutRouteProps<any>[] = [
             !!props.settingsCascade.final?.experimentalFeatures?.showSearchContext &&
             !!props.settingsCascade.final?.experimentalFeatures?.showSearchContextManagement,
     },
-    {
```
Comment (limitedmage) [inline client/web/src/routes.tsx:255]: Do we also need to make any changes to the routed in Go?

File: client/web/src/communitySearchContexts/o3de.tsx [hunk 3]
```diff
@@ -2,12 +2,12 @@ import React from 'react'
 
 import { SearchPatternType } from '@sourcegraph/shared/src/graphql-operations'
 
-import { RepogroupPage, RepogroupPageProps } from './RepogroupPage'
-import { RepogroupMetadata } from './types'
+import { CommunitySearchContextPage, CommunitySearchContextPageProps } from './CommunitySearchContextPage'
+import { CommunitySearchContextMetadata } from './types'
 
-export const o3de: RepogroupMetadata = {
+export const o3de: CommunitySearchContextMetadata = {
```
Comment (limitedmage) [inline client/web/src/communitySearchContexts/o3de.tsx:8]: I don't like that these are still hardcoded here. Are there any plans to move the code defined here to the context creation soon?

File: client/web/src/communitySearchContexts/routes.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+const StanfordCommunitySearchContextPage = lazyComponent(
+    () => import('./Stanford'),
+    'StanfordCommunitySearchContextPage'
+)
+const CncfCommunitySearchContextPage = lazyComponent(() => import('./cncf'), 'CncfCommunitySearchContextPage')
+
+// Hack! Hardcode these routes into cmd/frontend/internal/app/ui/router.go
+export const communitySearchContextsRoutes: readonly LayoutRouteProps<any>[] = [
+    {
+        path: '/kubernetes',
```
Comment (limitedmage) [inline client/web/src/communitySearchContexts/routes.tsx:32]: I don't like the URLs still being based on the root. Any plans to migrate these URLs to something like `/context/kubernetes`, with 301 redirects?

**Review by limitedmage — APPROVED** [review]
I think this is a good transitional step. However, I hope we move away from repogroup/community context metadata being hardcoded.

File: client/web/src/routes.tsx (hunk 2 above)
Comment (novoselrok) [reply client/web/src/routes.tsx:255]: No, the routes stayed the same, I just moved them to a better location. I didn't want them polluting the main app routes.

File: client/web/src/communitySearchContexts/o3de.tsx (hunk 3 above)
Comment (novoselrok) [reply client/web/src/communitySearchContexts/o3de.tsx:8]: As you've said, it's a transitional step. It's definitely not something we would like to maintain. We tried to this properly but we ran short on design resources to build it properly. So we decided that our priority should be replacing repogroup pages with community search context pages. That will allow to clean up repogroups from our product and give us some time to figure out how to best design the search context pages.

Once we improve the search context pages, it should be trivial to migrate community search context pages to regular search contexts pages (including the 301 redirects you mentioned below).

**Pull request merged**